File: Leetcode/3223.py
from collections import defaultdict, Counter


class Solution:
    def minimumLength(self, s: str) -> int:
        freq = Counter(s)
        ans = 0
        for v in freq.values():
            if v < 3:
                ans += v
            else:
                ans += 1 if v & 1 else 2
        return ans

File: Leetcode/3223.go
package Leetcode

func minimumLength(s string) int {
	counter := make(map[rune]int)
	for _, ch := range s {
		counter[ch]++
	}
	ans := 0
	for _, count := range counter {
		if count&1 != 0 {
			ans++
		} else {
			ans += 2
		}
	}
	return ans
}
